src/tss2-esys/esys_crypto_mbed.c

```diff
@@ -526,27 +532,37 @@ iesys_cryptmbed_pk_encrypt(TPM2B_PUBLIC * pub_tpm_key,
                            size_t * out_size, const char *label)
 {
     TSS2_RC r = TSS2_RC_SUCCESS;
+    int hash_id = 0;
 
     mbedtls_rsa_context rsa_context;
+#if MBEDTLS_VERSION_MAJOR > 2
+    mbedtls_rsa_init(&rsa_context);
+#endif
```

Comment:
move this to line 561 then we don't need separate blocks for > version 2

Reply:
yes that makes sense